### PR TITLE
Add edit state modal

### PR DESCRIPTION
**Pull request by constanzauanini**
Screen Shot 2022-09-06 at 11 34 43

### CODE REVIEW

File: src/components/accession2/EditStateModal.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        <Button label={strings.CANCEL} type='passive' onClick={onClose} priority='secondary' key='button-1' />,
+        <Button onClick={saveState} label={strings.UPDATE} key='button-2' />,
+      ]}
+    >
+      <Grid item xs={12} textAlign='left'>
+        <Grid item xs={12}>
+          <Select
+            label={strings.STATUS}
+            placeholder={strings.SELECT}
+            options={ACCESSION_2_STATES}
```
Comment (karthikbtf) [inline src/components/accession2/EditStateModal.tsx:61]: There's a table in the PRD explaining what edit states are valid. We may want to verify that. It's ok to leave it like this for now if we come back and follow up.
Basically, we cannot allow setting any state here, only some states.
See https://terraformation.atlassian.net/wiki/spaces/PM/pages/428114078/PRD+Seed+Management+updates+web+app#Accession-status-rules

File: src/types/Accession.ts [hunk 2]
```diff
@@ -9,3 +9,18 @@ export const ACCESSION_STATES = [
   'Withdrawn',
   'Nursery',
 ];
+
+export const ACCESSION_2_STATES = [
```
[the remark this answers is not in the document]
Comment (constanzauanini) [reply src/types/Accession.ts:13]: Ok, I won't merge this PR until yours is merged, so I update with main after

[the remark this answers is not in the document]
Comment (karthikbtf) [reply src/types/Accession.ts:13]: I merged the standlone PR which exports these states. When you rebase it should work fine.

**Review by karthikbtf — APPROVED** [review]

**Review by karthikbtf — APPROVED** [review]
Looks good, just one comment on accession being optional in props.

**Review by karthikbtf — APPROVED** [review]
Thanks 🎉

**Pull request merged**